helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -134,7 +137,10 @@
     // offline for more than this specified time period, and users call purge participant API,
     // then the node will be removed.
     // The unit is milliseconds.
-    OFFLINE_DURATION_FOR_PURGE_MS
+    OFFLINE_DURATION_FOR_PURGE_MS,
+
+    // Resource rebalance configuration for the super cluster which manages Helix controller resources.
+    SUPER_CLUSTER_REBALANCE_CONFIG
```

Comment:
This could be confusing. Because other sub clusters can have this entry. What does it mean to the sub cluster? If this is only designed for super cluster. Would this be meaningful in ClusterConfig?

I have concerns to add it as specific for super cluster IS rebalance strategy. We should make it generic. Either we add it as default rebalance strategy for existing cluster. Or let's make the the configurable passing down from the admin API every time.

Reply:
My argument is that a sub-cluster can be a supercluster at the same time. It is possible to do a cascading setup. So for a sub-cluster, the additional setup still means a supercluster configuration, which will be used when it acts as a supercluster.

About the first suggestion, are you talking about use WAGED as the default rebalancer for the sub-cluster resource directly? That was the original design. I just have a concern about backward compatibility. But if you think it is fine, I'm totally fine with this simpler solution.

Regarding the other option, I thought about modifying the activeCluster API to accept the optional parameters. But this means we will need to modify helix-rest and helix-front as well. And there is a backward compatibility issue. I would rather deprecate this API and restrict it to only use the general resource creation APIs for activating a sub-cluster.

Reply:
As discussed with @dasahcc offline, we will just modify the default rebalance option for the sub-clusters. This will simplify this PR quite a lot. 👍